datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -627,7 +628,7 @@ def _submit_gauges_from_summary(self, metric_name, metric, scraper_config, hostn
                     hostname=custom_hostname,
                 )
             else:
-                sample[self.SAMPLE_LABELS]["quantile"] = float(sample[self.SAMPLE_LABELS]["quantile"])
+                sample[self.SAMPLE_LABELS]["quantile"] = str(float(sample[self.SAMPLE_LABELS]["quantile"]))
```

Comment:
do we really needs to convert the float value to a string?

Reply:
same for the `"le"` below label actually 🤔

Reply:
@clamoriniere There is no impact on how the code behave, but later on in the code we are using `to_string()` [here](https://github.com/DataDog/integrations-core/pull/4157/files#diff-85fd7415439d33e481bee8605a392317R673) (actually, it should be called `to_native_text`) and it's only intended to handle bytes or unicode. It's not intended for float, `to_string` won't have any effect on float and the result will still be a float, but we should avoid passing anything other than bytes or unicode to it, that's why I made this str() conversion.

Other than that, since we are dealing with labels, it make semantically more sense to use strings instead of float. What do you think ?

Reply:
IIRC `sample[self.SAMPLE_LABELS]["quantile"]` is already a string maybe we should remove the `float` conversion in the first place ? Cannot remember why it's there in the first place, there might be a reason 🤔

Reply:
@mfpierre I think the reason might be to convert [`"3.1104e+07"`](https://github.com/DataDog/integrations-core/blob/9f6599742fdb604fcd21df38c3d0479fe4c9d555/datadog_checks_base/tests/test_openmetrics.py#L363) into [`"31104000.0"`](https://github.com/DataDog/integrations-core/blob/9f6599742fdb604fcd21df38c3d0479fe4c9d555/datadog_checks_base/tests/test_openmetrics.py#L370), but I don't have more context than that.